### PR TITLE
use DB connector instead of web API

### PR DESCRIPTION
**Pull request by nikooo777**
using the web API is not smart. this PR fixes that issue

### CODE REVIEW

File: server/utils/chainquery/index.js [hunk 1]
```diff
@@ -48,13 +51,13 @@ export async function claimSync () {
...
     while (!finished) {
-      let claimsResponse = await getClaimsSince(syncState.LastSyncTime, lastID, groupSize);
-      let claims = JSON.parse(claimsResponse).data;
+      let claims = await getClaimsSince(syncState.LastSyncTime, lastID, groupSize);
       status.info = 'addingClaimsToElastic';
       for (let claim of claims) {
         if (claim.value === null) {
           console.log(claim);
-          await logErrorToSlack('Failed to process claim ' + claim.claimId + ' due to missing value');
+          // await logErrorToSlack('Failed to process claim ' + claim.claimId + ' due to missing value');
```
Comment (tiger5226) [inline server/utils/chainquery/index.js]: I put this here intentionally. I need to know when they fail so I can make sure I fix any data in chainquery. We should however, make sure its not hardcoded and only production reports.

File: server/utils/chainquery/index.js [hunk 2]
```diff
@@ -168,33 +172,45 @@ function getBlockedOutpoints () {
...
+        let r = results[i];
+        claims.push({
+          id                : r.id,
+          name              : r.name,
+          channel           : r.channel,
+          bid_state         : r.bid_state,
+          effective_amount  : r.effective_amount,
+          certificate_amount: r.certificate_amount,
+          claimId           : r.claimId,
+          value             : r.value,
```
Comment (tiger5226) [inline server/utils/chainquery/index.js]: I don't know if this is right brother. I needs to be pushed as a json string for it to work properly. I would need to test this. However, I think you think this is pretty solid.

File: server/index.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ require('winston-daily-rotate-file');
 // Setup logging
 winston.remove(winston.transports.Console);
 winston.add(winston.transports.Console, { colorize: true, timestamp: true, prettyPrint: true });
-var slackAPIKey = process.env.SLACK_HOOK_URL;
-var mySlack = new slack(slackAPIKey, {});
+let slackAPIKey = process.env.SLACK_HOOK_URL;
+let mySlack = new slack(slackAPIKey, {});
```
Comment (kauffj) [inline server/index.js]: `const` is _technically_ more appropriate than `let` here but it doesn't really matter

File: server/utils/chainquery/index.js [hunk 4]
```diff
@@ -168,33 +172,45 @@ function getBlockedOutpoints () {
   });
 }
 
+let connection = null;
```
Comment (kauffj) [inline server/utils/chainquery/index.js]: these lines should probably go at top of file

File: server/utils/chainquery/index.js [hunk 5]
```diff
@@ -168,33 +172,45 @@ function getBlockedOutpoints () {
...
+const chainqueryConfig = require('../../../chainquery-config.json');
+
 function getClaimsSince (time, lastID, MaxClaimsInCall) {
+  if (connection === null) {
```
Comment (kauffj) [inline server/utils/chainquery/index.js]: A slightly better pattern would be:

`connection = getConnection()`

```
function getConnection() {
  //instantiate singleton here
}
```

File: server/utils/chainquery/index.js [hunk 6]
```diff
@@ -168,33 +172,45 @@ function getBlockedOutpoints () {
...
+    connection.query(query, function (err, results, fields) {
+      if (err) {
+        console.error(err);
         logErrorToSlack('[Importer] Error getting updated claims. ' + err);
-        reject(err);
-      });
+        return reject(err);
+      }
+      let claims = [];
+      for (let i = 0; i < results.length; i++) {
```
Comment (kauffj) [inline server/utils/chainquery/index.js]: Not strictly necessary but you can use `Array.map` for this:
```
const claims = results.map(r => {
          id                : r.id,
          name              : r.name,
          channel           : r.channel,
          bid_state         : r.bid_state,
          effective_amount  : r.effective_amount,
          certificate_amount: r.certificate_amount,
          claimId           : r.claimId,
          value             : r.value
})
```

**Comment by tzarebczan** [comment]
@nikooo777 what's the status of this PR? Sitting out here a while.

**Comment by nikooo777** [comment]
I have to find time to clean it up... I'll try to get that done asap.

**Pull request merged**